tests/user_examples/extra_src/shared/porous_solid_particles.h

```diff
@@ -0,0 +1,72 @@
+/* ------------------------------------------------------------------------- *
+ *                                SPHinXsys                                  *
+ * ------------------------------------------------------------------------- *
+ * SPHinXsys (pronunciation: s'finksis) is an acronym from Smoothed Particle *
+ * Hydrodynamics for industrial compleX systems. It provides C++ APIs for    *
+ * physical accurate simulation and aims to model coupled industrial dynamic *
+ * systems including fluid, solid, multi-body dynamics and beyond with SPH   *
+ * (smoothed particle hydrodynamics), a meshless computational method using  *
+ * particle discretization.                                                  *
+ *                                                                           *
+ * SPHinXsys is partially funded by German Research Foundation               *
+ * (Deutsche Forschungsgemeinschaft) DFG HU1527/6-1, HU1527/10-1,            *
+ *  HU1527/12-1 and HU1527/12-4.                                             *
+ *                                                                           *
+ * Portions copyright (c) 2017-2023 Technical University of Munich and       *
+ * the authors' affiliations.                                                *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain a *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.        *
+ *                                                                           *
+ * ------------------------------------------------------------------------- */
+/**
+ * @file 	solid_particles.h
+ * @brief 	This is the derived class of base particles.
+ * @author	Chi ZHang, Dong Wu and Xiangyu Hu
+ */
+
+#ifndef POROUS_SOLID_PARTICLES_H
+#define POROUS_SOLID_PARTICLES_H
+
+#include "solid_particles.h"
+#include "porous_media_solid.h"
+ 
+namespace SPH
+{
+/**
+ *	pre-claimed classes
+ */
+//class PorousMediaSolid;
+ 
+/**
+ * @class PorousMediaParticles
+ * @brief A group of particles with elastic body particle data.
+ */
+	class PorousMediaParticles : public ElasticSolidParticles
```

Comment:
we need to discuss how to add porous media to the code in its simplest but extendable way,